shared/feature/networkuser/src/main/java/com/epmedu/animeal/networkuser/data/repository/NetworkRepositoryImpl.kt

```diff
@@ -39,11 +38,24 @@ class NetworkRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun updateNetworkUserAttributes(profile: Profile, authRequestHandler: AuthRequestHandler) {
-        userAttributesAPI.updateUserAttributes(profileToAuthUserMapper.map(profile), authRequestHandler)
+    override suspend fun updateNetworkUserAttributes(
+        profile: Profile,
+        onSuccess: () -> Unit,
+        onError: (exception: Throwable) -> Unit
+    ) {
+        when (val result = userAttributesAPI.updateUserAttributes(profileToAuthUserMapper.map(profile))) {
+            is ApiResult.Success -> onSuccess()
```

Comment:
Please refactor callbacks `onSuccess` and `onError` using `ActionResult` and `ActionDelegate`